### ISSUE TITLE
Next lint doesn't give warning of unused dependencies in other places than pages/components

### ISSUE THREAD

**Issue by Harsh-Sharma25** [vercel/next.js#73497]
### Link to the code that reproduces this issue

https://github.com/Harsh-Sharma25/lint-issue-next

### To Reproduce

1. pnpm i
2. pnpm lint
& then if you run `pnpm eslint .` you will see the difference.

### Current vs. Expected behavior

When you run `pnpm lint` which in turn runs next lint it doesn't gives warning for missing dependencies in the hooks folder but if you define a hook inside pages then it does. `pnpm eslint .` seems to give proper warnings.

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 11 Home Single Language
  Available memory (MB): 16077
  Available CPU cores: 16
Binaries:
  Node: 20.17.0
  npm: 10.8.2
  Yarn: N/A
  pnpm: 9.14.2
Relevant Packages:
  next: 15.0.4-canary.37 // Latest available version is detected (15.0.4-canary.37).
  eslint-config-next: 15.0.3
  react: 19.0.0-rc-66855b96-20241106
  react-dom: 19.0.0-rc-66855b96-20241106
  typescript: 5.7.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Linting

### Which stage(s) are affected? (Select all that apply)

next dev (local), next build (local)

### Additional context

My original project is not on canary release, it is on 15.0.3 and it also doesn't use react 19 rc, which is when I encountered this issue. I've also tested with the eslint version: 9.15.0.

**Comment by JesseKoldewijn** [comment]
Next lint to my knowledge just calls eslint. Which makes me think in this case that this might be a eslint config issue.

**Comment by Harsh-Sharma25** [comment]
> Just had a look at the repro. Correct me if I'm wrong in this case but to my knowledge eslint v9+ only supports the new flat js config style and not the json based config anymore. You might want to try if this also shows up on the newer eslint config style.

Just made a new branch with es-lint 9 setup in it the issue is still the same [here](https://github.com/Harsh-Sharma25/lint-issue-next/tree/with-eslint-9)

**Comment by MartinCura** [comment]
May it be because by default next's configuration of eslint does not look into your `hooks/` dir? try changing `eslint.dirs` in your next config.
https://nextjs.org/docs/pages/api-reference/config/eslint#linting-custom-directories-and-files

**Comment by Harsh-Sharma25** [comment]
> May it be because by default next's configuration of eslint does not look into your `hooks/` dir? try changing `eslint.dirs` in your next config. https://nextjs.org/docs/pages/api-reference/config/eslint#linting-custom-directories-and-files

Yeah, it works like that but that means you would have to include all those dirs that you want to run lint on (which is every directory mostly) & in most of my projects I don't use src/ directory. Maybe next lint should also run on all dirs same as `eslint . `, if src/ dir is missing.

**Comment by marcoshernanz** [comment]
Thanks for the report. This was the documented behavior of `next lint`, which only checked `app`, `pages`, `components`, `lib`, and `src` by default. Other directories could be included with `--dir` or `eslint.dirs`.

`next lint` was deprecated in Next.js 15.5 and removed in Next.js 16 in favor of running ESLint directly. `eslint .` provides the repository-wide behavior requested here, so this is no longer actionable. Closing as no longer applicable.
